crates/sui-core/src/traffic_controller/mod.rs

```diff
@@ -330,7 +330,15 @@ async fn handle_spam_tally(
     metrics: Arc<TrafficControllerMetrics>,
     mem_drainfile_present: bool,
 ) -> Result<(), reqwest::Error> {
-    if !policy_config.spam_sample_rate.is_sampled().await {
+    let sampled = futures::future::join_all(vec![
+        tally.spam_weight.is_sampled(),
+        policy_config.spam_sample_rate.is_sampled(),
```

Comment:
I don't understand why this requires anything async or `join_all`

Reply:
Missed this, you're right the function called here doesn't need to be async

Reply:
Fixed